### PR TITLE
Add test and fix resolving [new Constructor] refs

### PR DESCRIPTION
**Pull request by keertip**

### CODE REVIEW

File: lib/src/model_utils.dart [hunk 1]
```diff
@@ -50,7 +51,16 @@ String replaceAllLinks(String str, String findMatchingLink(String input)) {
       }
       String codeRef = str.substring(index + _leftChar.length, end);
       if (codeRef != null) {
-        var link = findMatchingLink(codeRef);
+        var link;
+        // support for [new Constructor]
+        var refs = codeRef.split(' ');
+        if (refs.length == 2 && refs[0] == 'new') {
```
Comment (sethladd) [inline lib/src/model_utils.dart]: you can always do refs.first which might be cleaner

File: test/model_test.dart [hunk 2]
```diff
@@ -170,7 +170,7 @@ void main() {
 
     test('docs refs', () {
       expect(B.resolveReferences(),
-          'Extends class <a href="ex/Apple_class.html">Apple</a>');
+          'Extends class <a href="ex/Apple_class.html">Apple</a>, use new <a href="ex/Apple/Apple.html">Apple</a>');
```
Comment (sethladd) [inline test/model_test.dart]: I think we should include `new` in the link itself, making it clearer that it's a constructor we're linking to

Comment (keertip) [inline test/model_test.dart]: done.

File: lib/src/model_utils.dart (hunk 1 above)
Comment (keertip) [inline lib/src/model_utils.dart]: done.

File: lib/src/model.dart [hunk 3]
```diff
@@ -178,13 +178,18 @@ abstract class ModelElement {
       return docs;
     }
 
-    String _getMatchingLink(String codeRef) {
+    String _getMatchingLink(String codeRef, [bool isConstructor = false]) {
```
Comment (sethladd) [inline lib/src/model.dart]: sorry, should have caught this the first time. generally, booleans are better has named optional params. It's the difference between:

`foo(true)`

and

`foo(specialFlag: true)`

in readability.

Your call.

File: lib/src/model_utils.dart [hunk 4]
```diff
@@ -37,7 +37,8 @@ bool isInExportedLibraries(
     List<LibraryElement> libraries, LibraryElement library) => libraries
     .any((lib) => lib == library || lib.exportedLibraries.contains(library));
 
-String replaceAllLinks(String str, String findMatchingLink(String input)) {
+String replaceAllLinks(
+    String str, String findMatchingLink(String input, [bool isContructor])) {
```
Comment (sethladd) [inline lib/src/model_utils.dart]: did you want a default value here?

**Comment by sethladd** [comment]
lgtm with comments

File: lib/src/model.dart (hunk 3 above)
Comment (keertip) [inline lib/src/model.dart]: Did think about making it named, but was not sure of the convention. Will change it, more readable the better!

**Pull request merged**